libenv/sysinfo.c

```diff
@@ -1487,17 +1547,24 @@ static void OSClasses(EvalContext *ctx)
      * For example, when being run on either FreeBSD 10.0 or 10.1 a class
      * called freebsd_10 will be defined
      */
-    for (char *sp = vbuff; *sp != '\0'; sp++)
+    char *sp = NULL;
```

Comment:
The assignment to `NULL` is not necessary here since you assign it to something else on the next line.
```suggestion
    char *sp;
```